extensions/src/AWSSDK.Extensions.NETCore.Setup/ClientFactory.cs

```diff
@@ -26,24 +30,20 @@ namespace Amazon.Extensions.NETCore.Setup
     /// <summary>
     /// The factory class for creating AWS service clients from the AWS SDK for .NET.
     /// </summary>
-#if NET8_0_OR_GREATER
-    [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCode(Amazon.Extensions.NETCore.Setup.InternalConstants.RequiresUnreferencedCodeMessage)]
-#endif
-    internal class ClientFactory
+    internal class ClientFactory<T>
+        where T : IAmazonService
     {
         private static readonly Type[] EMPTY_TYPES = Array.Empty<Type>();
         private static readonly object[] EMPTY_PARAMETERS = Array.Empty<object>();
 
-        private Type _serviceInterfaceType;
         private AWSOptions _awsOptions;
 
         /// <summary>
         /// Constructs an instance of the ClientFactory
         /// </summary>
         /// <param name="type">The type object for the Amazon service client interface, for example IAmazonS3.</param>
-        internal ClientFactory(Type type, AWSOptions awsOptions)
+        internal ClientFactory(AWSOptions awsOptions)
```

Comment:
`param name="type"` this parameter was removed

Reply:
Fixed